### PR TITLE
FIO-8402: fixed an issue where Validation Triggering on initial Form load

### PR DESCRIPTION
**Pull request by KatrinKhilko**
## Link to Jira Ticket

https://formio.atlassian.net/browse/FIO-8402

## Description

**What changed?**

The issue was caused by calculated values, form load use onChange method that performs calculation and set pristine flag to false even if form wasn't initialized and user hasn't made any changes. Choose adding check for form initialization. 

## Dependencies

n/a

## How has this PR been tested?

Tested manually, all unit/integration tests pass locally

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [x] My changes generate no new warnings
- [ ] My changes include tests that prove my fix is effective (or that my feature works as intended)
- [x] New and existing unit/integration tests pass locally with my changes
- [ ] Any dependent changes have corresponding PRs that are listed above

### CODE REVIEW

File: src/components/_classes/component/Component.js [hunk 1]
```diff
@@ -2975,7 +2975,7 @@ export default class Component extends Element {
     this.calculatedValue = fastCloneDeep(calculatedValue);
 
     if (changed) {
-      if (!flags.noPristineChangeOnModified) {
+      if (!flags.noPristineChangeOnModified && this.root.initialized) {
```
Comment (brendanbond) [inline src/components/_classes/component/Component.js:2978]: Why can't we write tests for this? Even though it's a relatively simple change, a test or two would define the expected behavior of our program for developers and prevent regressions. It's also a good opportunity to test your assumptions to make sure there are no edge cases that haven't been considered.

**Review by brendanbond — CHANGES_REQUESTED** [review]

**Review by brendanbond — APPROVED** [review]
